src/gsy_e/models/strategy/forward/__init__.py

```diff
@@ -94,14 +119,122 @@ def event_offer_traded(self, *, market_id, trade):
     def event_bid_traded(self, *, market_id, bid_trade):
         """Method triggered by the MarketEvent.BID_TRADED event."""
 
+    @property
+    def fully_automated_trading(self):
+        """Is the strategy in fully automated trading mode."""
+        return self.enabled and self._create_order_updater
+
     def event_tick(self):
-        self._update_open_orders()
+        if self.enabled:
+            self._update_open_orders()
 
     def event_market_cycle(self):
         self._delete_past_order_updaters()
-        self._post_orders_to_new_markets()
+        if self.fully_automated_trading:
+            self._post_orders_to_new_markets()
 
     @property
     def state(self) -> "StateInterface":
         """Get the state class of the strategy. Needs to be implemented by all strategies"""
         raise NotImplementedError
+
+    def apply_live_event(self, event: Dict):
+        """Apply the incoming live event to the strategy."""
+        self._live_event_handler.dispatch(event)
+
+
+class ForwardLiveEvents:
+    """Handle live events for the forward strategies."""
+
+    def __init__(self, strategy):
+        self._strategy = strategy
+
+    def dispatch(self, event: Dict):
+        """Apply a live event to the strategy."""
+        if not B2BLiveEvents.is_supported_event(event.get("type")):
+            self._strategy.log.error(
+                "Invalid event (%s) for area %s.", event, self._strategy.area.name)
+            return
+
+        if event.get("type") == B2BLiveEvents.ENABLE_TRADING_EVENT_NAME:
+            self._auto_trading_event(event.get("args"))
+        if event.get("type") == B2BLiveEvents.DISABLE_TRADING_EVENT_NAME:
+            self._stop_auto_trading_event(event.get("args"))
+        if event.get("type") == B2BLiveEvents.POST_ORDER_EVENT_NAME:
+            self._post_order_event(event.get("args"))
+        if event.get("type") == B2BLiveEvents.REMOVE_ORDER_EVENT_NAME:
+            self._remove_order_event(event.get("args"))
+
+    def _auto_trading_event(self, args):
+        """
+        Enable energy trading, and create order updater for all open markets between the start and
+        end time.
+        """
+        if not LiveEventArgsValidator(
+                self._strategy.log.error).validate_start_trading_event_args(args):
+            return
```

Comment:
What do you think about adding a decorator that deals with this?

Reply:
This will not be that reusable, but I will give it a try to implement it as a decorator. Thanks!

Reply:
I see. In retrospect, I agree, your solution was fine. 😅  But keep the decorator. Thanks a lot for dealing with this!